### DISCUSSION TITLE
FDSNWS service requires authentication

### DISCUSSION THREAD

**Discussion by asenookei** [nikosT/Gisola#56] [General]
Dear @nikosT 

I am trying to retrieve stream data using FDSNWS, which requires a username and password for data access. The configuration is set up like this, but when I run it, I get the error below.

Stream:
  Service:
    - [FDSNWS, http://192.168.1.20:8081/fdsnws/dataselect/1/queryauth, username:#password]
    

error message : 

2024-09-26 07:18:52,214 INFO Retrieving waveforms based on inventory from 2024-09-23T22:53:47.613923Z to 2024-09-23T23:59:44.613923Z
2024-09-26 07:18:52,243 INFO No FDSN services could be discovered at 'http://192.168.1.20:8081/fdsnws/dataselect/1/queryauth'. This could be due to a temporary service outage or an invalid FDSN service address.
2024-09-26 07:18:52,244 INFO Saving waveforms in one mseed file in /home/sysop/Gisola/obs/results/2024/bmkg2024stvd/2024-09-26T07:18:38.201678Z/streams.mseed
2024-09-26 07:18:52,244 ERROR Fatal Error occurred

How should I configure the stream section if the FDSNWS service requires authentication?

2. I am trying to remove one sensor from the automated results using the command below. "IA" is the network, and "SBJI" is the sensor with channels SHZ, SHN, and SHE. However, after running the command, the sensor is still there, and the results have not changed from the automated process.

./gisola.py -c ../obs/config_auto.yaml -e bmkg2024sjbp --revise -s IA.SBJI --remove

What is the correct command to remove one or more sensors (for example, removing 3 sensors)? Both manual and automatic results are attached.

3. How do I identify poor-performing sensors from the list of sensors processed automatically and i will remove them and process the data manually to improve the quality of the results?

thanks,
[output.zip](https://github.com/user-attachments/files/17144751/output.zip)

**Comment by nikosT** [comment]
hi @asenookei ,

Concerning 1, the proper way is to use a TOKEN to access restricted data from dataselect. Then configuration would be similar to this:
```yaml
  Service:
    - [FDSNWS, http://192.168.1.20:8081, '/home/nikos/Downloads/eidatoken']
```
The filename MUST NOT have "." in file (e.g. eidatoken)
Also, perhaps http**s** must be used instead of http in the URL.


About 2, you have to provide only the name of the station and not the network too (see https://github.com/nikosT/Gisola/wiki/Usage).

About 3, I don't understand the question there. There are the functionality of the modules that you can enable to identify poor-performing sensors. See https://github.com/nikosT/Gisola/wiki/Modules. Usually, a good module is the ppsd. You could try the online case since the offline needs some setup (but it will take longer to calculate MT). You could also add the clip module. Moreover, the log file of an event, prints out which streams passed or not the modules check, therefore you can identify which are. If you need to modify the thresholds that determine bad or good events you can change the respective code inside the modules directory.

**Comment by esokos** [comment]
Actually understanding the "poor-performing sensors " (i guess you mean stations?) is not that easy. One thing you could check is the noise at the selected freq band of the inversion the other is to check for disturbances in data. Gisola has modules to do this but i would suggest to always have a look at the records yourself. What most users do is to look at the VR per trace and just remove the ones with low fit BUT this is not a safe option you should first understand WHY these are not fitted ... it could happen that you fit a problematic trace (e.g. it has large amplitudes and inversion will try to fit it) and then you don't fit the "good" ones ...
